cr-python-config-lib-master/foo/lib/config/configloader.py

```diff
@@ -0,0 +1,98 @@
+import os
+from typing import Optional
+
+import boto3
+import yaml
+
+
+class ConfigLoaderException(Exception):
+    pass
+
+
+class Config:
+
+    def __init__(self, config: dict, source: str):
+        self._config = config
+        self._source = source
+
+    @property
+    def source(self):
+        return self._source
+
+    def __repr__(self):
+        return (f'Config(config={self._config!r}'
+                f'       source={self.source!r})')
+
+    def __getattr__(self, name):
+        return self._config[name]
```

Comment:
I assume this should be:

def __getattr__(self, name):
        # Handles config.not_existing_key case
        # Here should be some error handling in case config.key does not exist

def __getattribute__(self, name):
        # Handles both config.not_existing_key and config.existing_key cases, the first one invokes __getattr__
        return self._config[name]